src/Sylius/Bundle/ApiBundle/DataProvider/ProductCollectionDataProvider.php

```diff
@@ -23,11 +25,13 @@ public function __construct(CollectionDataProviderInterface $collectionDataProvi
 
     public function getCollection(string $resourceClass, string $operationName = null, array $context = [])
     {
+        /** @var ChannelInterface $channel */
+        $channel = $context[ContextKeys::CHANNEL];
         $products = $this->collectionDataProvider->getCollection($resourceClass, $operationName, $context);
```

Comment:
Check getCollection method signature. I think $context is redunant argument

Reply:
As we have decoration here, this argument will finally get to DoctrineDataProvider:
here's a symfony-cast explanation https://prnt.sc/1sb9np9 (resource: https://symfonycasts.com/screencast/api-platform-extending/decorate-provider#play )

Reply:
Okey, thanks